cprover_bindings/src/goto_program/expr.rs

```diff
@@ -953,6 +955,9 @@ impl Expr {
                     || (lhs.typ.is_pointer() && rhs.typ.is_integer())
             }
             ROk => lhs.typ.is_pointer() && rhs.typ.is_c_size_t(),
+            VectorEqual | VectorNotequal | VectorGe | VectorLe | VectorGt | VectorLt => {
+                lhs.typ == rhs.typ && lhs.typ.is_vector()
```

Comment:
I think we should be more explicit about the conditions we expect here. All the conditions bellow must hold:
- `lhs.len() == rhs.len()`
- `lhs.typ.is_vector()`
- `lhs.typ.base_type().is_integer()`

Reply:
Done, but using this stricter condition:
```
        lhs.typ.is_vector()
            && lhs.typ == rhs.typ
            && lhs.typ.len() == ret_typ.len()
            && ret_typ.base_type().unwrap().is_integer()
```